### PR TITLE
.import.rio_xlsx dynamically discovers the formal arguments for read_…

### PR DESCRIPTION
**Pull request by bokov**
…xlsx or read.xlsx and discards all arguments in '...' that don't match, instead of having to catch them individually. Included tests. While I was at it, noticed there were no xls tests, so added a test file and tests (but left .import.rio_xls untouched so far)

See issue #221

### CODE REVIEW

File: R/import_methods.R [hunk 1]
```diff
@@ -320,13 +320,23 @@ function(file,
     }
     if (isTRUE(readxl)) {
         requireNamespace("readxl")
-        if ("rows" %in% names(a)) {
-            warning("'rows' argument ignored when readxl = TRUE. Use 'range' instead.")
-            a[["rows"]] <- NULL
-        }
+        frml <- formals(read_xlsx)
+        unused <- setdiff(names(a),names(frml))
+        if(length(unused)>0) {
```
Comment (leeper) [inline R/import_methods.R]: Can you format this per style guide:

if (length(unused) > 0) {

File: R/import_methods.R [hunk 2]
```diff
@@ -320,13 +320,23 @@ function(file,
...
+          warning("The following arguments were ignored when readxl = TRUE:",
+                  "\n",paste(unused,collapse=', '))}
```
Comment (leeper) [inline R/import_methods.R]: add spaces after commas; } on its own line

File: R/import_methods.R [hunk 3]
```diff
@@ -320,13 +320,23 @@ function(file,
     }
     if (isTRUE(readxl)) {
         requireNamespace("readxl")
-        if ("rows" %in% names(a)) {
-            warning("'rows' argument ignored when readxl = TRUE. Use 'range' instead.")
```
Comment (leeper) [inline R/import_methods.R]: Why delete this?

Comment (bokov) [reply R/import_methods.R]: The reason is that the code I added a few lines below...
```
if (length(unused)>0) {
          warning("The following arguments were ignored when readxl = TRUE:",
                  "\n", paste(unused, collapse=', '))
        }
```
...will already issue a warning for any invalid ignored arguments, _including but not limited to_ `rows`. So if somebody passes it a `rows` argument, they will get warned twice. 

That being said, I'll be happy to put the `if` statement back in if you want me to.

**Comment by bokov** [comment]
I have pushed the requested changes and standing by to your decision about deleted part.

While we're at it, quick question-- how do you feel about committing the following not directly related changes either here or in a new PR?

* New file, `rio.Rproj` to slightly help other contributors who use RStudio to comply with some project-wide style standards with no extra effort
* Adding lines to `.Rbuildignore` to ignore the RStudio project files.

Thanks.

**Comment by leeper** [comment]
@bokov I'm going to merge this. Thanks for the contribution! Happy to have you add an .Rproj file as a separate PR.

**Pull request merged**